deploy_nixos/main.tf

```diff
@@ -194,14 +194,16 @@ resource "null_resource" "deploy_nixos" {
       data.external.nixos-instantiate.result["out_path"],
       "${var.target_user}@${var.target_host}",
       var.target_port,
-      local.build_on_target,
-      local.ssh_private_key == "" ? "-" : local.ssh_private_key,
       "switch",
       var.delete_older_than,
       ],
       local.extra_build_args
     )
     command = "ignoreme"
+    environment = {
+      BUILD_ON_TARGET = local.build_on_target
+      SSH_PRIVATE_KEY = local.ssh_private_key == "" ? "-" : local.ssh_private_key
```

Comment:
While this will hide the key itself from output, a variable still marked as sensitive will cause the output to be suppressed. Might be worth wrapping this in [nonsensitive()](https://developer.hashicorp.com/terraform/language/functions/nonsensitive) although that would require bumping the minimum required terraform version from 0.12 to 0.15. Also there should be nothing stopping callers from wrapping the key in `nonsensitive()` when they pass it to this module.